includes/class-wc-stripe-order-helper.php

```diff
@@ -440,6 +475,202 @@ public function update_stripe_setup_intent( ?WC_Order $order = null, string $int
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		$order->delete_meta_data( self::META_STRIPE_CARD_ID );
+	}
+
+	/**
+	 * Gets the Stripe UPE payment type for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @return false|string|null
+	 */
+	public function get_stripe_upe_payment_type( ?WC_Order $order = null ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		return $order->get_meta( self::META_STRIPE_UPE_PAYMENT_TYPE, true );
+	}
+
+	/**
+	 * Updates the Stripe UPE payment type for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @param string $payment_type
+	 * @return false|void
+	 */
+	public function update_stripe_upe_payment_type( ?WC_Order $order = null, string $payment_type = '' ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		$order->update_meta_data( self::META_STRIPE_UPE_PAYMENT_TYPE, $payment_type );
+	}
+
+	/**
+	 * Gets the Stripe UPE waiting for redirect for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @return bool|null
+	 */
+	public function get_stripe_upe_waiting_for_redirect( ?WC_Order $order = null ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		return $order->get_meta( self::META_STRIPE_UPE_WAITING_FOR_REDIRECT, true );
+	}
+
+	/**
+	 * Updates the Stripe UPE waiting for redirect for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @param bool $waiting_for_redirect
+	 * @return false|void
+	 */
+	public function update_stripe_upe_waiting_for_redirect( ?WC_Order $order = null, bool $waiting_for_redirect = false ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		$order->update_meta_data( self::META_STRIPE_UPE_WAITING_FOR_REDIRECT, $waiting_for_redirect );
+	}
+
+	/**
+	 * Deletes the Stripe UPE waiting for redirect for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @return false|void
+	 */
+	public function delete_stripe_upe_waiting_for_redirect( ?WC_Order $order = null ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		$order->delete_meta_data( self::META_STRIPE_UPE_WAITING_FOR_REDIRECT );
+	}
+
+	/**
+	 * Gets the Stripe UPE redirect processed for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @return bool|null
+	 */
+	public function get_stripe_upe_redirect_processed( ?WC_Order $order = null ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		return $order->get_meta( self::META_STRIPE_UPE_REDIRECT_PROCESSED, true );
+	}
+
+	/**
+	 * Updates the Stripe UPE redirect processed for order.
+	 *
+	 * @since 10.0.0
+	 *
+	 * @param WC_Order|null $order
+	 * @param bool $redirect_processed
+	 * @return false|void
+	 */
+	public function update_stripe_upe_redirect_processed( ?WC_Order $order = null, bool $redirect_processed = false ) {
+		if ( is_null( $order ) ) {
+			return false;
+		}
+
+		$order->update_meta_data( self::META_STRIPE_UPE_REDIRECT_PROCESSED, $redirect_processed );
+	}
```

Comment:
Not blocking, but I am wondering whether it might be worth creating some internal/private helper functions so we have less duplicated/boilerplate code. Something like the following is what I was thinking, but I realise that it adds one more layer of indirection.
```suggestion
	public function update_stripe_upe_redirect_processed( ?WC_Order $order = null, bool $redirect_processed = false ) {
		return $this->_update_order_meta( $order, self::META_STRIPE_UPE_REDIRECT_PROCESSED, $redirect_processed );
	}

	private function _update_order_meta( ?WC_Order $order,  string $meta_key, $new_value ) {
		if ( null === $order ) {
			return false;
		}
		
		$order->update_meta_data( $meta_key, $new_value );
	}
```

Reply:
I actually like the idea! Implemented it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4705/commits/aff3db9c4f49931d8c6c2a7d5c4b578a6b47b177